SPHINXsys/src/shared/particle_dynamics/solid_dynamics/thin_structure_math.cpp

```diff
@@ -243,5 +243,30 @@ namespace SPH
 			return corrected_almansi_strain;
 		}
 		//=================================================================================================//
+		Mat2d getCorrectionMatrix(const Mat2d &local_deformation_part_one)
+		{
+			Mat2d correction_matrix(0.0);
+			correction_matrix[0][0] = 1.0 / (local_deformation_part_one[0][0] + TinyReal);
+			return correction_matrix;
+		}
+		//=================================================================================================//
+		Mat3d getCorrectionMatrix(const Mat3d &local_deformation_part_one)
+		{
+			Mat2d reduced_local_deformation_part_one;
+			reduced_local_deformation_part_one[0][0] = local_deformation_part_one[0][0];
+			reduced_local_deformation_part_one[0][1] = local_deformation_part_one[0][1];
+			reduced_local_deformation_part_one[1][0] = local_deformation_part_one[1][0];
+			reduced_local_deformation_part_one[1][1] = local_deformation_part_one[1][1];
+			Mat2d reduced_correction_matrix; 
+			reduced_correction_matrix = SimTK::inverse(reduced_local_deformation_part_one); 
+			Mat3d correction_matrix(0.0); 
```

Comment:
Is it fully safe to not have 1 on the diagonal at `correction_matrix[2][2]` ?

Reply:
Because the shell model is a dimension reduction model, the correction_matrix[2][2] should be 0, not 1.